### PR TITLE
WT-7670 Modify test tag format and tag additional python tests.

### PR DESCRIPTION
**Pull request by luke-pearson**
I will continue to tag more python tests @etienneptl, I was hoping you could review my python script changes in the meantime.

I thought about making the `test_coverage.md` an html file instead so we can support merging of cells and improve overall readability however that is not the purpose of this ticket.

### CODE REVIEW

File: dist/test_tag.py [hunk 1]
```diff
@@ -21,22 +22,39 @@
...
+valid_tags = []
 
 END_TAG = "[END_TAGS]"
 IGNORE_FILE = "ignored_file"
-NB_TAG_ARGS = 3
+NB_TAG_ARGS = 2
 START_TAG = "[TEST_TAGS]"
 #####
 
+##### FUNCTIONS ####
```
Comment (etienneptl) [inline dist/test_tag.py:33]: Missing `#` on the right side 👀

File: dist/test_tag.py [hunk 2]
```diff
@@ -89,56 +107,61 @@
...
                 continue
 
         # Check if end of test tag
         if END_TAG in line:
             # END_TAG should not be before START_TAG
-            if is_start == False:
-                print("Error syntax in file " + filename + ". Unexpected tag: " + END_TAG)
-                exit()
+            if in_tag_block == False:
+                print("Syntax Error in file: " + filename + ". Unexpected tag: " + END_TAG)
```
Comment (etienneptl) [inline dist/test_tag.py:120]: Change `Syntax Error` to `Syntax error`

File: dist/test_tag.py [hunk 3]
```diff
@@ -89,56 +107,61 @@
...
-                print("Tag is not valid ! Add the new tag to test_tags.ok:\n" + line)
-                exit()
-            else:
-                is_file_tagged = True
+
+            # Validate the tag's correctness.
+            tag_count = validate_tag(tag, filename)
+            is_file_tagged = True
+
+            # Add the test type to the tag if it wasn't already.
```
Comment (etienneptl) [inline dist/test_tag.py:154]: What is this special handling for ? Why didnt we have it before if it is necessary ?

Comment (luke-pearson) [reply dist/test_tag.py:154]: Essentially to save tags from being too verbose its assumed that .c files are "data_correctness" test types and ".py" is a "functional_correctness" test type.

Comment (etienneptl) [reply dist/test_tag.py:154]: Assuming that we will never have a Python script that is a data_correctness test and vice versa

Comment (luke-pearson) [reply dist/test_tag.py:154]: The logic I'm using here is wrong as a valid tag could be 1, 2 or 3, in length. I don't know if theres a clean way to check this. I'll on it for a bit.

Comment (luke-pearson) [reply dist/test_tag.py:158]: Alright I've pushed up a change that does it a bit better.

File: dist/test_tag.py (hunk 2 above)
Comment (luke-pearson) [reply dist/test_tag.py]: 866d36d72

File: dist/test_tag.py (hunk 1 above)
Comment (luke-pearson) [reply dist/test_tag.py]: 866d36d72

File: dist/test_tag.py [hunk 4]
```diff
@@ -89,56 +104,67 @@
...
                 continue
-            # Check if current tag is valid
-            if not line in valid_tags:
-                print("Tag is not valid ! Add the new tag to test_tags.ok:\n" + line)
-                exit()
-            else:
-                is_file_tagged = True
+
+            # Validate the tag's correctness.
+            tag_count = validate_tag(tag, filename)
```
Comment (etienneptl) [inline dist/test_tag.py:148]: `tag_count` does not seem used anymore. If we dont need it, we might as well return a bool from `validate_tag`

File: dist/test_tag.py [hunk 5]
```diff
@@ -55,16 +70,17 @@
 ##### RETRIEVE VALID TAGS #####
 validation_file = open("test_tags.ok", "r")
 
-# The file has the following pattern
-# <COMPONENT>:<TESTING_TYPE>:<TESTING_AREA>:<DESCRIPTION>
-# A tag is made of the three first values: COMPONENT, TEST_TYPE and TESTING_AREA
+# The validation file contains a list of valid tag words.
```
Comment (etienneptl) [inline dist/test_tag.py:73]: Should we say it's one per line ? That's the format expected.

File: dist/test_tag.py [hunk 6]
```diff
@@ -89,56 +104,67 @@
...
+
+            # Validate the tag's correctness.
+            tag_count = validate_tag(tag, filename)
+            is_file_tagged = True
+
+            skip_adding_test_type = False
+            # Add the test type to the tag if it wasn't already.
+            for test_type_tag in test_type_tags:
+                if (tag.startswith(test_type_tag)):
+                    skip_adding_test_type = True
```
Comment (etienneptl) [inline dist/test_tag.py:155]: You can add `break` after this line even though `test_type_tags` has only 2 values for now.

File: dist/test_tag.py (hunk 5 above)
Comment (luke-pearson) [reply dist/test_tag.py]: bbdb81c9c

File: dist/test_tag.py (hunk 4 above)
Comment (luke-pearson) [reply dist/test_tag.py]: bbdb81c9c

File: dist/test_tag.py (hunk 6 above)
Comment (luke-pearson) [reply dist/test_tag.py:154]: bbdb81c9c

**Review by etienneptl — APPROVED** [review]
LGTM, great work !

**Pull request merged**